src/client/channelOwner.ts

```diff
@@ -84,11 +84,11 @@ export abstract class ChannelOwner<T extends channels.Channel = channels.Channel
     };
   }
 
-  async _wrapApiCall<T>(apiName: string, func: () => Promise<T>, logger?: Logger): Promise<T> {
+  async _wrapApiCall<T, K extends channels.Channel>(apiName: string, func: (channel: K) => Promise<T>, logger?: Logger): Promise<T> {
```

Comment:
```async _wrapApiCall<R>(apiName: string, func: (channel: T) => Promise<R>, logger?: Logger): Promise<R> {```

I believe this should infer the channel type from `ChannelOwner<T>` and you would not have to write `channel: channels.FooChannel` everywhere.